_i18n/en/_docs/best-practices.md

```diff
@@ -56,3 +56,33 @@ Interceptor.attach(f, {
   }
 });
 {% endhighlight %}
+
+### Reusing arguments
+
+When reading the arguments on the *onEnter* callback, it is common to access each argument by their index. But what happens when an argument is accessed multiple times? Take for example this line of code:
+
+{% highlight javascript %}
+Interceptor.attach(f, {
+  onEnter: function (args) {
+    if (!args[0].readUtf8String(4).includes('MZ'))
+    {
+      console.log(args[0].readUtf8String());
+    }
+  }
+});
+{% endhighlight %}
+
+In the above example the first argument is obtained from the *args* array twice and this is paying a cost of requesting *frida-gum* for this information twice. To avoid querying and paying the performance toll when operating with the same argument multiple times, it is best to store this information using *this*:
+
+
+{% highlight javascript %}
+Interceptor.attach(f, {
+  onEnter: function (args) {
+    this.firstArg = args[0];
+    if (!this.firstArg.readUtf8String(4).includes('MZ'))
+    {
+      console.log(this.firstArg.readUtf8String());
```

Comment:
I noticed the old samples still have the old syntax, updated those.

Does `const` inside the `onEnter` callback ensure that it live throughout the callback or is `this` required? When checking the previous examples it was using `var` so I switched them to const at least

Reply:
Also, modified the example to call a different API `hexdump` to forcibly re-use `args[0]`

Reply:
No, `const` just declares it as a variable that can't be re-assigned. But things should only be assigned to `this` if needed in `onLeave`, or if the lifetime (e.g. in case of `Memory.alloc*()`) needs to be anchored to the duration of the call.